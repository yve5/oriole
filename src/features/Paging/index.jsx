const Paging = () => (
  <div className="card mt-3">
    <div className="card-header">Paging</div>

    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        <table className="table">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">First</th>
              <th scope="col">Last</th>
              <th scope="col">Handle</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row">1</th>
              <td>Mark</td>
              <td>Otto</td>
              <td>@mdo</td>
            </tr>
            <tr>
              <th scope="row">2</th>
              <td>Jacob</td>
              <td>Thornton</td>
              <td>@fat</td>
            </tr>
            <tr>
              <th scope="row">3</th>
              <td colSpan="2">Larry the Bird</td>
              <td>@twitter</td>
            </tr>
          </tbody>
        </table>

        <nav aria-label="Page navigation">
          <ul className="pagination">
            {['Previous', '1', '2', '3', 'Next'].map((item) => (
              <li className="page-item" key={item}>
                <button className="page-link" type="button">
                  {item}
                </button>
              </li>
            ))}
          </ul>
        </nav>
      </li>
    </ul>
  </div>
);

export default Paging;
