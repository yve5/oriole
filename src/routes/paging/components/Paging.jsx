import React from 'react';
import { I18n } from '../../../shared/i18n';

const Paging = () => (
  <div className="card">
    <h1>
      <I18n>Hello World</I18n>
    </h1>

    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        <table className="table">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">First</th>
              <th scope="col">Last</th>
              <th scope="col">Handle</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row">1</th>
              <td>Mark</td>
              <td>Otto</td>
              <td>@mdo</td>
            </tr>
            <tr>
              <th scope="row">2</th>
              <td>Jacob</td>
              <td>Thornton</td>
              <td>@fat</td>
            </tr>
            <tr>
              <th scope="row">3</th>
              <td colSpan="2">Larry the Bird</td>
              <td>@twitter</td>
            </tr>
          </tbody>
        </table>

        <nav>
          <ul className="pagination justify-content-center">
            <li className="page-item">
              <button className="page-link" type="button">
                <span aria-hidden="true">&laquo;</span>
              </button>
            </li>
            {[1, 2, 3].map((item) => (
              <li className="page-item" key={item}>
                <button className="page-link" type="button">
                  {item}
                </button>
              </li>
            ))}
            <li className="page-item">
              <button className="page-link" type="button">
                <span aria-hidden="true">&raquo;</span>
              </button>
            </li>
          </ul>
        </nav>
      </li>
    </ul>
  </div>
);

export default Paging;
