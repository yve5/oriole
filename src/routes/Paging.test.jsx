import React from 'react';
import { expect, test, vi } from 'vitest';
import { act, create } from 'react-test-renderer';

import Paging from './Paging';

test('should match expected snapshot', () => {
  HTMLCanvasElement.prototype.getContext = vi.fn(() => ({
    arc: vi.fn(),
    beginPath: vi.fn(),
    clearRect: vi.fn(),
    closePath: vi.fn(),
    fill: vi.fn(),
    lineTo: vi.fn(),
    moveTo: vi.fn(),
    rotate: vi.fn(),
    restore: vi.fn(),
    save: vi.fn(),
    scale: vi.fn(),
    translate: vi.fn(),
  }));

  const component = create(<Paging />);

  act(() => {
    component.update(<Paging />);
  });

  expect(component.toJSON()).toMatchSnapshot();
});
