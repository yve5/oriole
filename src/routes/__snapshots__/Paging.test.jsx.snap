// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match expected snapshot 1`] = `
<div
  className="card"
>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              First
            </th>
            <th
              scope="col"
            >
              Last
            </th>
            <th
              scope="col"
            >
              Handle
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              scope="row"
            >
              1
            </th>
            <td>
              Mark
            </td>
            <td>
              Otto
            </td>
            <td>
              @mdo
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              2
            </th>
            <td>
              Jacob
            </td>
            <td>
              Thornton
            </td>
            <td>
              @fat
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            >
              3
            </th>
            <td
              colSpan="2"
            >
              Larry the Bird
            </td>
            <td>
              @twitter
            </td>
          </tr>
        </tbody>
      </table>
      <nav>
        <ul
          className="pagination"
        >
          <li
            className="page-item"
          >
            <button
              className="page-link"
              type="button"
            >
              Previous
            </button>
          </li>
          <li
            className="page-item"
          >
            <button
              className="page-link"
              type="button"
            >
              1
            </button>
          </li>
          <li
            className="page-item"
          >
            <button
              className="page-link"
              type="button"
            >
              2
            </button>
          </li>
          <li
            className="page-item"
          >
            <button
              className="page-link"
              type="button"
            >
              3
            </button>
          </li>
          <li
            className="page-item"
          >
            <button
              className="page-link"
              type="button"
            >
              Next
            </button>
          </li>
        </ul>
      </nav>
    </li>
  </ul>
</div>
`;
